src/commands/data/import/bulk.ts

```diff
@@ -0,0 +1,246 @@
+        },
+        {
+          label: 'Job Id',
+          type: 'dynamic-key-value',
+          bold: true,
+          get: (data) =>
+            data?.id &&
+            terminalLink(
+              data.id,
+              `${baseUrl}/lightning/setup/AsyncApiJobStatus/page?address=${encodeURIComponent(`/${data.id}`)}`
+            ),
+        },
+      ],
+    });
+
+    if (async) {
+      ms.goto('Creating ingest job');
+
+      const job = await createIngestJob(conn, flags.file, flags.sobject, flags['line-ending']);
+
+      ms.goto('Creating ingest job', job.getInfo());
+      ms.stop();
+
+      const cache = await BulkImportRequestCache.create();
+      await cache.createCacheEntryForRequest(job.id, conn.getUsername(), conn.getApiVersion());
+
+      this.log(messages.getMessage('export.resume', [job.id]));
+
+      return {
+        jobId: job.id,
+      };
+    }
+
+    // synchronous flow
+    const job = await createIngestJob(conn, flags.file, flags.sobject, flags['line-ending']);
+
+    ms.goto('Processing the job');
+
+    job.on('inProgress', (res: JobInfoV2) => {
+      ms.goto('Processing the job', res);
+    });
+
+    try {
+      await job.poll(5000, timeout.milliseconds);
+
+      const jobInfo = job.getInfo();
+
+      // send last data update so job status/num. of records processed/failed represent the last update
+      ms.goto('Processing the job', jobInfo);
+
+      if (jobInfo.numberRecordsFailed) {
+        ms.stop('failed');
+        // TODO: replace this msg to point to `sf data bulk results` when it's added (W-12408034)
+        throw messages.createError('error.failedRecordDetails', [
+          jobInfo.numberRecordsFailed,
+          conn.getUsername(),
+          job.id,
+        ]);
+      }
+
+      ms.stop();
+
+      return {
+        jobId: jobInfo.id,
+        processedRecords: jobInfo.numberRecordsProcessed,
+        successfulRecords: jobInfo.numberRecordsProcessed - (jobInfo.numberRecordsFailed ?? 0),
+        failedRecords: jobInfo.numberRecordsFailed,
+      };
+    } catch (err) {
+      const jobInfo = await job.check();
+
+      // send last data update so job status/num. of records processed/failed represent the last update
+      ms.goto('Processing the job', jobInfo);
+
+      if (err instanceof Error && err.name === 'JobPollingTimeout') {
+        ms.stop('paused');
+        throw messages.createError('error.timeout', [timeout.minutes, job.id]);
+      }
+
+      if (jobInfo.state === 'Failed') {
+        ms.stop('failed');
+        throw messages.createError(
+          'error.jobFailed',
+          [jobInfo.errorMessage, conn.getUsername(), job.id],
+          [],
+          err as Error
+        );
+      }
+
+      if (jobInfo.state === 'Aborted') {
+        ms.stop('failed');
+        // TODO: replace this msg to point to `sf data bulk results` when it's added (W-12408034)
+        throw messages.createError('error.jobAborted', [conn.getUsername(), job.id], [], err as Error);
```

Comment:
aborted jobs don't include an error msg, we log the `sf org open ..` msg so they can view more details in the org.